include/onnxruntime/core/framework/ort_value.h

```diff
@@ -75,6 +75,22 @@ struct OrtValue {
 #endif
   }
 
+  bool IsInvalid() const noexcept {
+    return (type_ != nullptr && type_->IsInvalidType());
+  }
+
+  bool IsNonTensor() const noexcept {
+    return (type_ != nullptr && type_->IsNonTensorType());
+  }
```

Comment:
Would not `!IsTensor()` be enough?

Reply:
I can remove that. I used them to improve an error message telling one output was not a tensor and I wondered what it could be. It turned out it is not possible to have optional output in custom ops. The output was just empty.

Reply:
removed